src/esp/nav/PathFinder.cpp

```diff
@@ -1128,6 +1132,42 @@ bool PathFinder::Impl::isNavigable(const vec3f& pt,
   return true;
 }
 
+std::vector<std::vector<bool>> PathFinder::Impl::getTopDownView(
+    const float res) {
+  std::pair<vec3f, vec3f> mapBounds = bounds();
+  vec3f bound1 = mapBounds.first;
+  vec3f bound2 = mapBounds.second;
+
+  float width = std::abs(bound1[0] - bound2[0]);
+  float height = std::abs(bound1[2] - bound2[2]);
+  int widthResolution = width / res;
+  int heightResolution = height / res;
+  float startx = fmin(bound1[0], bound2[0]);
+  float starty = fmin(bound1[2], bound2[2]);
+  float navigableHeight = getRandomNavigablePoint()[1];
+  std::vector<std::vector<bool>> topdownMap;
+
+  float curHeight = starty;
+  float curWidth = startx;
+  for (int h = 0; h < heightResolution; h++) {
+    std::vector<bool> curRow;
+    for (int w = 0; w < widthResolution; w++) {
+      vec3f point = vec3f(curWidth, navigableHeight, curHeight);
+      if (isNavigable(point, 0.5)) {
```

Comment:
This will need to change for the Eigen matrix, but anyways, you can do `curRow.push_back(isNavigable(point, 0.5))` here.